.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,59 @@
+name: Integration Tests
+
+on:
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build_and_test_triton_shared:
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - name: Checkout Triton
+      uses: actions/checkout@v4
+      with:
+        repository: 'openai/triton'
+        ref: '215b2e77a1d92f907bc4addfa3c3ba204028c32e' # known good commit "Add Shared Middle Layer to Triton via Plug-In"
```

Comment:
Probably not required right now, but I wonder if we want this in a file or something so we can script the checkout for local devs to make sure they are on the correct commit. This is the approach ONNX-MLIR uses for example https://github.com/onnx/onnx-mlir/blob/main/utils/clone-mlir.sh

Reply:
yeah, not the best place for this; but i'm okay with this for now.
I wonder if Phil wouldn't mind us adding some labels...